samza-api/src/main/java/org/apache/samza/table/remote/TablePart.java

```diff
@@ -44,4 +44,18 @@
     return Collections.emptyMap();
   }
 
+  /**
+   * Set tableId for this table part. This method can be called by sub-classes of
+   * {@link org.apache.samza.table.descriptors.BaseTableDescriptor}, where tableId is available.
+   * @param tableId table id
+   */
+  default void setTableId(String tableId) {
+  }
+
+  /**
+   * Return the tableId that has been set. Return empty string by default.
+   */
+  default String getTableId() {
+    return "";
```

Comment:
is throwing an exception better than an default value? 
Then the user is not allowed to get the id if it's not set.